Add files via upload

oniram22 opened the pull request:
See issue #39 for background information on what's being submitted here.  Thanks.

danjagnow commented on samples/__init__.py:
```diff
@@ -0,0 +1,836 @@
+
+            Args:
+                node_name(string): A node name to be used for the newly created node object.
+                ip_address(string): The ip address that is associated with the supplied node name.
```
I'd capitalize "IP" in the description.

danjagnow commented on samples/__init__.py:
```diff
@@ -0,0 +1,836 @@
+            return ""
+
+    def get_node_uri(self, node_name):
+        """ Returns the NodeURI for the given NodeName.  Uses a Swis query to the Solarwinds database to retrieve this
```
"Swis" should be "SWIS" in the comments.

danjagnow commented on samples/__init__.py:
```diff
@@ -0,0 +1,836 @@
+            return ""
+
+    def get_group_uri(self, group_name):
+        """ Returns the ContainerUrifor the given Group Name.  Uses a Swis query to the Solarwinds database to retrieve this
```
space before "for"

oniram22 commented:
I agree with all the recommended edit’s.  I corrected them and submitted a new pull request with the changes.

Michael J. Marino
Assistant Director of Network Services
Morgan, Lewis & Bockius LLP
1701 Market Street | Philadelphia, PA 19103-2921
Direct: +1.215.963.4658<tel:+1.215.963.4658> | Main: +1.215.963.5000<tel:+1.215.963.5000> | Fax: +1.215.963.5001<tel:+1.215.963.5001>
michael.marino@morganlewis.com<mailto:michael.marino@morganlewis.com> | www.morganlewis.com<http://www.morganlewis.com/>

From: Dan Jagnow <notifications@github.com>
Sent: Wednesday, October 23, 2019 3:22 PM
To: solarwinds/orionsdk-python <orionsdk-python@noreply.github.com>
Cc: Marino, Michael J. <michael.marino@morganlewis.com>; Author <author@noreply.github.com>
Subject: Re: [solarwinds/orionsdk-python] Add files via upload (#40)

[EXTERNAL EMAIL]

@danjagnow approved this pull request.

This is a tremendous bit of work! I've added a few minor comments about naming, but this looks like a great addition whether or not those tweaks get made.

________________________________

In samples/__init__.py<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_solarwinds_orionsdk-2Dpython_pull_40-23discussion-5Fr338220346&d=DwMCaQ&c=wbMekZ1iboz3wtx3lILI8YgCUSSh7g3G58syakvKORs&r=r5tlIKDVEYwQW03CNkZwUAK5jdzPocjIQyWgaykAjNI&m=dnRu5G5PBJhCD9UpoqFYCliXIpHENCKRkiFPGFBXR9Q&s=dFNlIhT6Vrq0hq6lnCkE_FEBjoIIWMF-aIAKDHjYIZA&e=>:

> @@ -0,0 +1,836 @@



+# -*- coding: utf-8 -*-



+"""Solarwinds Class Overview



"Solarwinds" should probably be "SolarWinds" with a capital 'W' throughout, for consistency with the brand.

________________________________

In samples/__init__.py<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_solarwinds_orionsdk-2Dpython_pull_40-23discussion-5Fr338220895&d=DwMCaQ&c=wbMekZ1iboz3wtx3lILI8YgCUSSh7g3G58syakvKORs&r=r5tlIKDVEYwQW03CNkZwUAK5jdzPocjIQyWgaykAjNI&m=dnRu5G5PBJhCD9UpoqFYCliXIpHENCKRkiFPGFBXR9Q&s=U5G1qGa7y-meMCF5aX_gohaHU0-y3gsrj0_DKP_DIBw&e=>:

> @@ -0,0 +1,836 @@



+# -*- coding: utf-8 -*-



+"""Solarwinds Class Overview



+



+The following class is used to create an object representing a Solarwinds NPM instance.  The object provides simple



Maybe "SolarWinds Orion" instead of "Solarwinds NPM"?

________________________________

In samples/__init__.py<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_solarwinds_orionsdk-2Dpython_pull_40-23discussion-5Fr338221239&d=DwMCaQ&c=wbMekZ1iboz3wtx3lILI8YgCUSSh7g3G58syakvKORs&r=r5tlIKDVEYwQW03CNkZwUAK5jdzPocjIQyWgaykAjNI&m=dnRu5G5PBJhCD9UpoqFYCliXIpHENCKRkiFPGFBXR9Q&s=9cQYVZlzha54RpBRknWf1Qp4CTaD7cQQGN7hnIQnS1s&e=>:

> @@ -0,0 +1,836 @@



+# -*- coding: utf-8 -*-



+"""Solarwinds Class Overview



+



+The following class is used to create an object representing a Solarwinds NPM instance.  The object provides simple



+getter and setter methods for common Solarwinds actions.  These methods abstract out the underlying details and SQL



This should probably be "SWQL" instead of "SQL", though it is true that it usually gets translated down to SQL in the bowels of SWIS.

________________________________

In samples/__init__.py<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_solarwinds_orionsdk-2Dpython_pull_40-23discussion-5Fr338226175&d=DwMCaQ&c=wbMekZ1iboz3wtx3lILI8YgCUSSh7g3G58syakvKORs&r=r5tlIKDVEYwQW03CNkZwUAK5jdzPocjIQyWgaykAjNI&m=dnRu5G5PBJhCD9UpoqFYCliXIpHENCKRkiFPGFBXR9Q&s=jc8c-AmYVKhvp3mTVaL4bQNEht-D_1Rm1819D3p8tlw&e=>:

> +                                   caption=node_name)



+



+        self.logger.info("get_node_uri - node uri query results: %s.", node_uri)



+        if node_uri['results']:



+            return node_uri['results'][0]['Uri']



+        else:



+            return ""



+



+    def add_node_using_snmp_v3(self, node_name, ip_address, snmpv3_username, snmpv3_priv_method, snmpv3_priv_pwd,



+                               snmpv3_auth_method, snmpv3_auth_pwd):



+        """ Creates a new node using the supplied name an ip address.  Configure with our standard SNMPv3 credentials.



+            Once created, attached all of the standard Cisco pollers.



+



+            Args:



+                node_name(string): A node name to be used for the newly created node object.



+                ip_address(string): The ip address that is associated with the supplied node name.



I'd capitalize "IP" in the description.

________________________________

In samples/__init__.py<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_solarwinds_orionsdk-2Dpython_pull_40-23discussion-5Fr338226467&d=DwMCaQ&c=wbMekZ1iboz3wtx3lILI8YgCUSSh7g3G58syakvKORs&r=r5tlIKDVEYwQW03CNkZwUAK5jdzPocjIQyWgaykAjNI&m=dnRu5G5PBJhCD9UpoqFYCliXIpHENCKRkiFPGFBXR9Q&s=Y4QcZNQbF-VsUVdEmq5hrI6YpDKDo-CWO3RaoXPuzuk&e=>:

> +                node_id (string): The node ID that corresponds to the specified node name.



+



+        """



+



+        node_id = self.swis.query("SELECT NodeID, Caption FROM Orion.Nodes WHERE Caption = @caption",



+                                  caption=node_name)



+



+        self.logger.info("get_node_id - node_id query results: %s.", node_id)



+



+        if node_id['results']:



+            return node_id['results'][0]['NodeID']



+        else:



+            return ""



+



+    def get_node_uri(self, node_name):



+        """ Returns the NodeURI for the given NodeName.  Uses a Swis query to the Solarwinds database to retrieve this



"Swis" should be "SWIS" in the comments.

________________________________

In samples/__init__.py<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_solarwinds_orionsdk-2Dpython_pull_40-23discussion-5Fr338230023&d=DwMCaQ&c=wbMekZ1iboz3wtx3lILI8YgCUSSh7g3G58syakvKORs&r=r5tlIKDVEYwQW03CNkZwUAK5jdzPocjIQyWgaykAjNI&m=dnRu5G5PBJhCD9UpoqFYCliXIpHENCKRkiFPGFBXR9Q&s=_jmre1xPx3JSLJ9fxX7EX966wahRpLr3g_LH-f3pPmE&e=>:

> +            Returns:



+                group_id (string): The group ID that corresponds to the specified group name.



+



+        """



+



+        group_id = self.swis.query("SELECT ContainerID FROM Orion.Container WHERE Name = @group_name",



+                                   group_name=group_name)



+        self.logger.info("get_group_id - group_id query results: %s", group_id)



+



+        if group_id['results']:



+            return group_id['results'][0]['ContainerID']



+        else:



+            return ""



+



+    def get_group_uri(self, group_name):



+        """ Returns the ContainerUrifor the given Group Name.  Uses a Swis query to the Solarwinds database to retrieve this



space before "for"

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_solarwinds_orionsdk-2Dpython_pull_40-3Femail-5Fsource-3Dnotifications-26email-5Ftoken-3DADINOH3K37ZQ6K374TUOEILQQCP37A5CNFSM4I6VRGQKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCI7O7NY-23pullrequestreview-2D306114487&d=DwMCaQ&c=wbMekZ1iboz3wtx3lILI8YgCUSSh7g3G58syakvKORs&r=r5tlIKDVEYwQW03CNkZwUAK5jdzPocjIQyWgaykAjNI&m=dnRu5G5PBJhCD9UpoqFYCliXIpHENCKRkiFPGFBXR9Q&s=X42F8XWOMi9GoveJwF0nCF-nGF6yN2yvtoNmERJyRxA&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ADINOH7MFAZ6AZPP5L5X7CTQQCP37ANCNFSM4I6VRGQA&d=DwMCaQ&c=wbMekZ1iboz3wtx3lILI8YgCUSSh7g3G58syakvKORs&r=r5tlIKDVEYwQW03CNkZwUAK5jdzPocjIQyWgaykAjNI&m=dnRu5G5PBJhCD9UpoqFYCliXIpHENCKRkiFPGFBXR9Q&s=zX4-SroZ6gvVGGXv89eMWoHrYh8RAoCkXCyyRBA0DoA&e=>.

DISCLAIMER
This e-mail message is intended only for the personal use
of the recipient(s) named above.  This message may be an
attorney-client communication and as such privileged and
confidential and/or it may include attorney work product.
If you are not an intended recipient, you may not review,
copy or distribute this message.  If you have received this
communication in error, please notify us immediately by
e-mail and delete the original message.
